### DISCUSSION TITLE
Laravel 11 Compatibility

### DISCUSSION THREAD

**Discussion by CrispMind** [zofe/rapyd-livewire#5] [Ideas]
Hello! 

Laravel 11 has been around for a while. Any chance rapyd-livewire could be updated to install into Laravel 11 projects? Currently there's a conflict with illuminate/contracts required by rapyd-livewire but illuminate/contracts has been replaced by laravel/framework in Laravel 11. 

Being able to use your amazing library with Laravel 11 would be above awesome!

Thanks!

**Comment by zofe** [comment]
There is a rewrite in progress: https://github.com/zofe/rapyd-admin
I work on it when I have free time.

I would like to condense it into a new rapyd-livewire library, adding a set of modules ready to generate extended apps.

At the moment I don't feel like suggesting its adoption.
Neither do I have in “production” applications that use it.

However, I did just start by updating the rapyd-livewire codebase to make it compatible with laravel 11.
If you have an application in production I suggest you fork rapyd-livewire and try and copy the changes I have already applied to the rapyd-admin codebase.

If you are creating a new application or can afford to change dependencies, take a look at rapyd-admin (which is for laravel 11)

**Reply by CrispMind** [reply]
Ah, that's great news! Thanks for working on that.

I ended up creating a fork as you suggested and tweaking it it work with Laravel 11 and Livewire 3. The latter forced me to drop Vue components though as livewire-vue is not compatible with Livewire 3 (and won't ever be it seems) but luckily my app can live with that. The bundled Alpine.js had to go too as it's now part of Livewire 3. Otherwise it's holding up really well so far.

**Reply by zofe** [reply]
in rapyd-admin I am also giving up vuejs, and I don't mind it too much. What I like about livewire is the testability of the components, and the fewer javascript frameworks there are in the way the easier it is to write and maintain tests
